features/minimal-printf/README.md

```diff
@@ -0,0 +1,115 @@
+# Minimal printf and snprintf
+
+Library supports both printf and snprintf in 1252 bytes of flash.
+
+Prints directly to stdio/UART without using malloc. All flags and precision modifiers are ignored.
+Floating point is enabled by default.
+Printing to a FILE stream is enabled by default.
+
+Supports:
+* %d: signed integer [h, hh, (none), l, ll, z, j, t].
+* %i: signed integer [h, hh, (none), l, ll, z, j, t].
+* %u: unsigned integer [h, hh, (none), l, ll, z, j, t].
+* %x: unsigned integer [h, hh, (none), l, ll, z, j, t], printed as hexadecimal number (e.g., ff).
+* %X: unsigned integer [h, hh, (none), l, ll, z, j, t], printed as hexadecimal number (e.g., FF).
+* %f: floating point (disabled by default).
+* %F: floating point (disabled by default, treated as %f).
+* %g: floating point (disabled by default, treated as %f).
+* %G: floating point (disabled by default, treated as %f).
+* %c: character.
+* %s: string.
+* %p: pointer (e.g. 0x00123456).
+
+Unrecognized format specifiers are treated as ordinary characters.
+
+Floating point support:
+* Floating point is enabled by default.
+* All floating points are treated as %f.
+* No support for inf, infinity or nan
+
+## Usage
+
+To replace the standard implementation of the printf functions with the ones in this library:
+
+Compile with mbed-cli using the custom `mprintf` profile. For example, to compile in release mode:
+
+```
+$ mbed compile -t <toolchain> -m <target> --profile release --profile mprintf
+```
+
+## Enabling FILE stream, floating point and 64 bit integers
+
+In mbed_app.json:
+
+```
+    "target_overrides": {
+        "*": {
+            "minimal-printf.enable-file-stream": true,
+            "minimal-printf.enable-floating-point": true,
+            "minimal-printf.set-floating-point-max-decimals": 6,
+            "minimal-printf.enable-64-bit": true
+        }
+    }
+```
+
+
+## Size comparison
+
+Example application compiled with minimal mbed OS (no RTOS) using this .mbedignore:
```

Comment:
Could we try to measure it for normal mode blinky and separately for bare metal blinky rather then the mbed ignore?

Reply:
Done. See https://github.com/ARMmbed/mbed-os/blob/dc53154ae3449f3f0cfad3e08a48cea2c13aa71d/platform/minimal-printf/README.md